### ISSUE TITLE
DataTable: row group header colspan doesn't span full row

### ISSUE THREAD

**Issue by phoomin2012** [primefaces/primevue#6496]
### Describe the bug

When using row group with subheader mode, the colspan for the row group header doesn't span to full row like pictures below.

<img width="50%" alt="DataTable Row Group" src="https://github.com/user-attachments/assets/016cc76f-ed96-4059-8b5b-0a395b441ace">
<img width="50%" alt="DataTable Row Group with Column Group" src="https://github.com/user-attachments/assets/1e4accce-1631-4e42-a2a4-0906402f754a">

I found that this line of code will calculate colspan by decreasing the amount of column by 1. After removing `- 1`, the row group header seems to render correctly.
https://github.com/primefaces/primevue/blob/433232cf97a42debb33769d93e5762edbfdc4d4c/packages/primevue/src/datatable/BodyRow.vue#L4

### Reproducer

https://stackblitz.com/edit/primevue-4-vite-issue-template-ucphaz?file=src%2FApp.vue

### PrimeVue version

4.0.7

### Vue version

3.x

### Language

ALL

### Build / Runtime

Vue CLI App

### Steps to reproduce the behavior

See Stackblitz and inspect row group header.
First table row group header colspan is 3 when it should be 4.
Second table row group header colspan is 5 when it should be 6.

### Expected behavior

colspan for group headers should span the full row.

**Comment by phoomin2012** [comment]
## Solution
After I investigated more, I got that we must add `Column` with atrribute `field` value same as `groupRowsBy` in `DataTable` like this.

```html
<DataTable
    :value="data"
    row-group-mode="subheader"
    group-rows-by="group.id"
    ....
>
   <Column field="group.id" header="Representative"></Column> <-- This column will hide automatically
   .....
</DataTable>
```

The document won't said anything about this (but include in example code). I hope that this should write down on document too.

**Comment by tugcekucukoglu** [comment]
Thanks for the updates.

**Comment by acrepaz-oc** [comment]
Hi, I've also came across a very similar issue.
When using the DataTable RowGroup functionality with nested objects for the fields, it will generate an incorrect row-span. It is always one less than desired.

For example,

```
 <DataTable
          rowGroupMode="subheader"
          groupRowsBy="type.name"
```

```
      <Column
        field="type.name"
```

The above will generate a group header, but if I have 5 visible columns, the rowspan will be set to 4.


**Workaround**
I've found a workaround for this is to not use nested fields, and instead update my data array to flatten the value.
eg, `type.name` to `typeName`.

This then works as desired with the correct rowspan.

```
 <DataTable
          rowGroupMode="subheader"
          groupRowsBy="typeName"
```

```
      <Column
        field="typeName"
```
